src/apps/api/tests/test_competition_participants.py

```diff
@@ -46,3 +48,13 @@ def test_normal_user_cannot_see_participants(self):
         resp = self.get_participants()
         assert CompetitionParticipant.objects.filter(competition=self.comp).count() > 0
         assert len(resp.json()) == 0
+
+    def test_cant_participate_multiple_times(self):
+        self.client.login(username='norm', password='norm')
+        resp = self.client.post(reverse('competition-register', kwargs={"pk": self.comp.pk}))
+        assert resp.status_code == 201
+        CompetitionParticipant.objects.get(user=self.norm)
+        resp = self.client.post(reverse('competition-register', kwargs={"pk": self.comp.pk}))
+        assert resp.status_code == 400
+        assert "You already applied for participation in this competition!" in resp.data[0]
+        assert len(CompetitionParticipant.objects.filter(user=self.norm)) == 1
```

Comment:
for better optimization and accuracy
```suggestion
        assert CompetitionParticipant.objects.filter(user=self.norm, competition=self.comp).count() == 1
```

Reply:
✔️